src/pages/guides/javascript/how-to-add-metadata-to-spl-tokens.md

```diff
@@ -0,0 +1,318 @@
+
+// Airdrop 2 SOL to the identity
+// if you end up with a 429 too many requests error, you may have to use
+// the a different rpc other than the free default one supplied.
+await umi.rpc.airdrop(umi.identity.publicKey, sol(2));
+```
+
+### Use an Existing Wallet Stored Locally
+
+```ts
+const umi = createUmi("https://api.devnet.solana.com")
+	.use(mplTokenMetadata())
+	.use(mplToolbox());
+
+// You will need to us fs and navigate the filesystem to
+// load the wallet you wish to use via relative pathing.
+const walletFile = const imageFile = fs.readFileSync('./keypair.json')
+
+// Convert your walletFile onto a keypair.
+let keypair = umi.eddsa.createKeypairFromSecretKey(new Uint8Array(walletFile));
+
+// Load the keypair into umi.
+umi.use(keypairIdentity(umiSigner));
+```
+
+## Creating the Token
+
+This section is can be skipped if you already have an SPL token initialized, otherwise let's update the `createSPLToken` wrapper function. 
+
+We will use the `createMint` method from `mpl-toolbox` library to initialize an SPL token mint. We will also need to initialize a new keypair for the mint account.
+
+```typescript
+// initialize a new keypair for our mint account
+const mint = generateSigner(umi);
+
+// Create SPL token wrapper function
+async function createSPLToken() {
+	let tx = await createMint(umi, {
+		mint,
+		decimals: 6,
+		mintAuthority: umi.identity.publicKey,
+		freezeAuthority: null,
+	}).sendAndConfirm(umi);
+
+	let txSig = base58.deserialize(tx.signature);
+
+	console.log("token mint address -> ", mint.publicKey);
+	console.log(`https://explorer.solana.com/tx/${txSig}?cluster=devnet`);
+}
+```
+
+Calling the function will initialize a new SPL token, but it won't have any metadata associated with it.
+
+## Adding Metadata
+
+Adding metadata is also as simple as creating the token, we will utilize the `createMetadataAccountV3` method from the `mpl-token-metadata` library.
+
+Also note that this guide assumes that you already had your off-chain token metadata prepared beforehand. We will need the name, off-chain uri address and symbol
+
+```json
+name: "Solana Gold",
+symbol: "GOLDSOL",
+uri: "https://raw.githubusercontent.com/solana-developers/program-examples/new-examples/tokens/tokens/.assets/spl-token.json",
+```
+
+```typescript
+// Sample Metadata for our Token
+const tokenMetadata = {
+	name: "Solana Gold",
+	symbol: "GOLDSOL",
+	uri: "https://raw.githubusercontent.com/solana-developers/program-examples/new-examples/tokens/tokens/.assets/spl-token.json",
+};
+
+// Add metadata to an existing SPL token wrapper function
+async function addMetadata() {
+    // derive the metadata account that will store our metadata data onchain
+	const metadataAccountAddress = await findMetadataPda(umi, {
+		mint: mint.publicKey,
+	});
+
+   // add metadata to our already initialized token using `createMetadataAccountV3` 
+   // from the token metadata program
+	const tx = await createMetadataAccountV3(umi, {
```

Comment:
we generally recommend to use the createV1 Instruction. createMetadataAccountV3 is deprecated as mentioned in the developer hub here https://developers.metaplex.com/token-metadata/mint#creating-mint-and-metadata-accounts